lib/rspec/matchers/built_in/include.rb

```diff
@@ -139,6 +139,10 @@ def diff_would_wrongly_highlight_matched_item?
             actual.include?(str) && lines.none? { |line| line == str }
           end
         end
+
+        def convert_to_hash?(obj)
+          !obj.is_a?(::Hash) && obj.respond_to?(:to_hash)
```

Comment:
Please use `!(::Hash === obj)` instead of `!obj.is_a?(::Hash)`.   The reason is that `is_a?` can be defined on any object to do anything -- or can even be undefined on some objects.  For example, `BasicObject` does not define `is_a?`:

```
2.4.4 :003 > BasicObject.new.is_a?(::Hash)
NoMethodError: undefined method `is_a?' for #<BasicObject:0x007fa910e077f0>
	from (irb):3
	from /usr/bin/irb:11:in `<main>'
```

In contrast, `!(::Hash === obj)` will work with _any_ object and we can count on its behavior being consistent unless someone monkey patches `Hash.===`.  Generally speaking we don't support environments where core types are monkey patched to misbehave but we do need to support users passing us arbitrary objects that define their own behavior.

Reply:
I see. Thank you. I changed it.